tools/shaderc/shaderc.cpp

```diff
@@ -1071,8 +1071,10 @@ namespace bgfx
 		return word;
 	}
 
-	bool compileShader(const char* _varying, const char* _comment, char* _shader, uint32_t _shaderLen, Options& _options, bx::WriterI* _writer)
+	bool compileShader(const char* _varying, const char* _comment, char* _shader, uint32_t _shaderLen, Options& _options, bx::WriterI* _writer, bx::WriterI* _messages)
 	{
+		bx::Error messageErr;
```

Comment:
If you're not expecting/handling errors, use `bx::ErrorAssert messageErr`.


And since there are two writers it would be good to be explicitly named:

`bx::WriterI* _shaderWriter, bx::WriterI* _messageWriter`

Apply these rules in all other places you modified.